app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/RepeatPasscodeFragment.kt

```diff
@@ -144,7 +144,10 @@ class RepeatPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>(
             findNavController().popBackStack(R.id.createPasscodeFragment, true)
         }
 
-        binding.input.hideSoftKeyboard()
+        if(isAdded) {
```

Comment:
I would catch this issue earlier and prevent dismissCreatePasscodeFragment() from being triggered if fragment is not added. This check could be used in state observer
```kotlin
lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)
```
how come PasscodeViewModel.PasscodeSetup is observed in disposed fragment?

Reply:
The CreatePasscodeFragment is still visible. And needs to be dismissed. We also need to show the confirmation that the passcode was setup. 
But the keyboard is not visible when coming back from the locked screen. So it doesn't need to be hidden.